### PR TITLE
Projects can now be updated by clicking the 'update' button on admin panel

### PR DESCRIPTION
**Pull request by snack-attack**

### CODE REVIEW

File: addProject.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+require_once './database.php';
+
+if(!empty($_POST['title']) && !empty($_POST['description'])) {
```
Comment (mporam) [inline addProject.php]: PSR2

File: addProject.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    $title = filter_var($_POST['title'], FILTER_SANITIZE_STRING);
+    $description = filter_var($_POST['description'], FILTER_SANITIZE_STRING);
+    if(!empty($_POST['site_url'])) {
```
Comment (mporam) [inline addProject.php]: PSR2

File: addProject.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    
+    $sql = "INSERT INTO `project` (`title`, `description`, `site_url`, `code_url`) 
+            VALUES (:title, :description, :site_url, :code_url);";
+    $query = $db->prepare($sql);
+    $query->bindParam(':title', $title); 
+    $query->bindParam(':description', $description);
+    $query->bindParam(':site_url', $site_url); 
+    $query->bindParam(':code_url', $code_url); 
+
+    $result = $query->execute();
```
Comment (mporam) [inline addProject.php]: not doing anything with `$result`

File: admin.php [hunk 4]
```diff
@@ -40,10 +41,11 @@
                         echo '<tr><td data-label="Title">';
                         echo $project['title'];
                         echo '</td>
-                        <td>
-                        <button class="edit">Edit</button>
-                        <button class="delete">Delete</button>
-                        </td></tr>';
+                            <td>
+                                <a href="projectForm.php?id=' . $project['id']; echo '">Edit</a>
```
Comment (mporam) [inline admin.php]: why add another echo? just concat on the end:
```
?id=' . $project['id'] . '">Edit
```

File: editProject.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+include_once './database.php';
+
+if(!empty($_GET)) {
```
Comment (mporam) [inline editProject.php]: PSR2!!!!

Comment (mporam) [reply editProject.php]: also you want to check specifically for `$_GET['id']`

File: editProject.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        WHERE `id`=:id;";
+
+        $query = $db->prepare($sql);
+        $query->bindParam(':id', $id);
+        $query->bindParam(':title', $title); 
+        $query->bindParam(':description', $description);
+        $query->bindParam(':site_url', $site_url); 
+        $query->bindParam(':code_url', $code_url); 
+    
+        $result = $query->execute();
```
Comment (mporam) [inline editProject.php:27]: no check on $result

File: projectForm.php [hunk 7]
```diff
@@ -1,3 +1,21 @@
+<?php
+    include_once 'database.php';
+
+    $addProject = 'addProject.php';
+    $editProject = 'editProject.php';
+
+    if(!empty($_GET)) {
```
Comment (mporam) [inline projectForm.php]: check for get id

File: projectForm.php [hunk 8]
```diff
@@ -16,19 +34,47 @@
             <p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="#">upgrade your browser</a> to improve your experience.</p>
         <![endif]-->
         <h1>Create a New Project</h1>
-        <form method="POST" action="editProject.php">
+        <form method="POST" action="<?php if(!empty($_GET)) { echo $editProject . '?id=' . $project['id']; } else { echo $addProject; }; ?>">
```
Comment (mporam) [inline projectForm.php]: you could do this by having one file with an if to check if there is an id sent

File: projectForm.php [hunk 9]
```diff
@@ -16,19 +34,47 @@
...
-                <input type="text" placeholder="Enter Project Title" name="title" required>
+                <label for="title">Project Title</label>
+                <input type="text" placeholder="Enter Project Title" name="title" value="<?php 
+                    if(!empty($_GET)) { 
+                        echo $project['title'];}  
+                  ?>" required>
+
+                <label for="description">Description</label>
+                <textarea placeholder="Enter Description" name="description" required><?php
+                    if(!empty($_GET)) {
```
Comment (mporam) [inline projectForm.php]: you can use the null coalesce operator here

File: projectForm.php [hunk 10]
```diff
@@ -16,19 +34,47 @@
...
+                ?></textarea>
 
-                <label for="description"><b>Description</b></label>
-                <textarea placeholder="Enter Description" name="description" required></textarea>
+                <label for="site_url">Site URL</label>
+                <input type="url" placeholder="Enter Site URL" name="site_url" value="<?php
+                    if(!empty($_GET)) {
+                        echo $project['site_url'];
+                    }
+                ?>
```
Comment (mporam) [inline projectForm.php]: close php immediately before the " or you will get whitespace at the end

File: projectForm.php [hunk 11]
```diff
@@ -16,19 +34,47 @@
...
+                        echo $project['code_url'];
+                    }
+                ?>
+                "></input>
 
-                <label for="code_url"><b>Code URL</b></label>
-                <input type="url" placeholder="Enter Site URL" name="code_url" required></input>
+                <!-- <label for="tags">Tags</label>
+                <?php
+                $sql = "SELECT `id`, `tag`.`name` as name FROM `tag`;";
```
Comment (mporam) [inline projectForm.php]: please remove if your no longer using tags

**Review by mporam — CHANGES_REQUESTED** [review]

File: projectForm.php (hunk 8 above)
Comment (snack-attack) [reply projectForm.php]: it feels nicer to split it out? But I am planning on refactoring to a class in order to get stuff lookin better - lots of repeat codeeeeee

File: admin.php [hunk 12]
```diff
@@ -42,8 +42,8 @@
                         echo $project['title'];
                         echo '</td>
                             <td>
-                                <a href="projectForm.php?id=' . $project['id']; echo '">Edit</a>
-                                <a href="projectForm.php?id=' . $project['id']; echo '" id="delete">Delete</a>
+                                <a href="projectForm.php?id='.$project['id'].'">Edit</a>
+                                <a href="projectForm.php?id='.$project['id'].'" id="delete">Delete</a>
```
Comment (mporam) [inline admin.php]: PSR2 says put spaces around concatenation operators:
```
?id=' . $project['id'] . '">Edit
```

File: projectForm.php [hunk 13]
```diff
@@ -4,7 +4,7 @@
     $addProject = 'addProject.php';
     $editProject = 'editProject.php';
 
-    if(!empty($_GET)) {
+    if(!empty($_GET['id'])) {
```
Comment (mporam) [inline projectForm.php]: PSR2

File: projectForm.php [hunk 14]
```diff
@@ -34,47 +34,19 @@
             <p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="#">upgrade your browser</a> to improve your experience.</p>
         <![endif]-->
         <h1>Create a New Project</h1>
-        <form method="POST" action="<?php if(!empty($_GET)) { echo $editProject . '?id=' . $project['id']; } else { echo $addProject; }; ?>">
+        <form method="POST" action="<?php if (!empty($_GET['id'])) { echo $editProject . '?id=' . $project['id']; } else { echo $addProject; }; ?>">
```
Comment (mporam) [inline projectForm.php]: use a ternarary here

**Review by mporam — CHANGES_REQUESTED** [review]

File: projectForm.php [hunk 15]
```diff
@@ -34,7 +34,7 @@
             <p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="#">upgrade your browser</a> to improve your experience.</p>
         <![endif]-->
         <h1>Create a New Project</h1>
-        <form method="POST" action="<?php if (!empty($_GET['id'])) { echo $editProject . '?id=' . $project['id']; } else { echo $addProject; }; ?>">
+        <form method="POST" action="<?= $editProject . '?id=' . $project['id'] ?? $addProject; ?>">
```
Comment (mporam) [inline projectForm.php]: this doesnt work

**Review by mporam — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**